catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/PollerRunner.java

```diff
@@ -0,0 +1,203 @@
+      final String message = "Unable to start polling because scheduledExecutorService is shutdown";
+      LOGGER.warn("{}. Try restarting.", message);
+      throw new IllegalStateException(message);
+    }
+
+    startPolling(pollIntervalMinutes);
+  }
+
+  /**
+   * @throws IllegalArgumentException if {@code initialPollIntervalMinutes} is less than {@value
+   *     #MINIMUM_POLL_INTERVAL_MINUTES}
+   * @throws IllegalStateException if unable to schedule polling
+   */
+  private synchronized void startPolling(final long pollIntervalMinutes) {
+    validPollIntervalMinutes(pollIntervalMinutes);
+
+    if (scheduledPollingFuture != null) {
+      scheduledPollingFuture.cancel(true);
+      LOGGER.debug(
+          "Stopped the scheduled process to poll all of the current sources at a fixed rate");
+    }
+
+    // From the ScheduledExecutorService#scheduleAtFixedRate javadoc: "If any execution of this task
+    // takes longer than its period, then subsequent executions may start late, but will not
+    // concurrently execute."
+    try {
+      scheduledPollingFuture =
+          scheduledExecutorService.scheduleAtFixedRate(
+              () -> poll(pollIntervalMinutes), 0, pollIntervalMinutes, TimeUnit.MINUTES);
+    } catch (final RejectedExecutionException e) {
+      final String message =
+          "Unable to schedule polling at at a fixed rate of " + pollIntervalMinutes + " minute(s)";
+      LOGGER.warn("{}. Try restarting.", message);
+      throw new IllegalStateException(message, e);
+    }
+    LOGGER.debug(
+        "Successfully scheduled a process to poll all of the current sources at a fixed rate of {} minute(s), where polls will time out at {} minute(s)",
+        pollIntervalMinutes,
+        pollIntervalMinutes);
+  }
+
+  @SuppressWarnings("squid:S1181" /*Catching throwable intentionally*/)
+  private void poll(final long pollIntervalMinutes) {
+    final ImmutableMap<K, Callable<V>> current = getCurrent();
+    final int currentSourcesCount = current.size();
+
+    try {
+      LOGGER.trace(
+          "Starting a process to poll the {} current source(s), where polls will time out at {} minute(s)",
+          currentSourcesCount,
+          pollIntervalMinutes);
+      // If there are any unhandled exceptions in this method, all future calls will be suppressed.
+      poller.pollCurrent(pollIntervalMinutes, TimeUnit.MINUTES, current);
+      LOGGER.trace(
+          "Successfully finished the process of polling the {} source(s)", currentSourcesCount);
+    } catch (InterruptedException e) {
+      LOGGER.debug("Interrupted while waiting for polling to complete", e);
+      LOGGER.warn(
+          "A scheduled poll failed. See debug log for more details. All of the current will continue to be polled at a fixed rate of {} minutes. Try restarting if future polls also fail.",
+          pollIntervalMinutes);
+      // TODO How should the InterruptedException be handled here?
```

Comment:
We probably should assume the system or executor are being shut down and stop everything.

Reply:
How about this:
```java
LOGGER.warn(
    "A scheduled poll was interrupted. Destroying the ScheduledExecutorService. Future polls are no longer scheduled. Try restarting the system.",
    pollIntervalMinutes);
destroy();
Thread.currentThread().interrupt();
```
?